services/mt/run_translate.go

```diff
@@ -0,0 +1,196 @@
+
+	api "github.com/Smartling/api-sdk-go/api/mt"
+	sdkfile "github.com/Smartling/api-sdk-go/helpers/sm_file"
+	"github.com/reconquest/hierr-go"
+)
+
+// TranslateParams is the parameters for the RunTranslate method.
+type TranslateParams struct {
+	SourceLocale     string
+	DetectLanguage   bool
+	TargetLocales    []string
+	InputDirectory   string
+	OutputDirectory  string
+	Directives       map[string]string
+	Progress         bool
+	OverrideFileType string
+	FileOrPattern    string
+	ProjectID        string
+	AccountUID       api.AccountUID
+}
+
+func (s service) RunTranslate(ctx context.Context, p TranslateParams) ([]TranslateOutput, error) {
+	var res []TranslateOutput
+
+	base, pattern := globfiles.GetDirectoryFromPattern(p.FileOrPattern)
+	files, err := globfiles.LocallyFunc(
+		p.InputDirectory,
+		base,
+		pattern,
+	)
+	if err != nil {
+		return nil, clierror.NewError(
+			hierr.Errorf(
+				err,
+				`unable to find matching files to upload`,
+			),
+
+			`Check, that specified pattern is valid and refer to help for`+
+				` more information about glob patterns.`,
+		)
+	}
+
+	if len(files) == 0 {
+		return nil, clierror.NewError(
+			fmt.Errorf(`no files found by specified patterns`),
+
+			`Check command line pattern if any and configuration file for`+
+				` more patterns to search for.`,
+		)
+	}
+
+	base = filepath.Dir(base)
+
+	for _, file := range files {
+		name, err := filepath.Abs(file)
+		if err != nil {
+			return nil, clierror.NewError(
+				hierr.Errorf(
+					err,
+					`unable to resolve absolute path to file: %q`,
+					file,
+				),
+
+				`Check, that file exists and you have proper permissions `+
+					`to access it.`,
+			)
+		}
+
+		if relPath, err := filepath.Rel(p.InputDirectory, name); err != nil || strings.HasPrefix(relPath, "..") {
+			return nil, clierror.NewError(
+				errors.New(
+					`you are trying to push file outside project directory`,
+				),
+				`Check file path and path to configuration file and try again.`,
+			)
+		}
+
+		name, err = filepath.Rel(p.InputDirectory, name)
+		if err != nil {
+			return nil, clierror.NewError(
+				hierr.Errorf(
+					err,
+					`unable to resolve relative path to file: %q`,
+					file,
+				),
+
+				`Check, that file exists and you have proper permissions `+
+					`to access it.`,
+			)
+		}
+
+		contents, err := os.ReadFile(file)
+		if err != nil {
+			return nil, clierror.NewError(
+				hierr.Errorf(
+					err,
+					`unable to read file contents "%s"`,
+					file,
+				),
+
+				`Check that file exists and readable by current user.`,
+			)
+		}
+		request := sdkfile.FileUploadRequest{
+			File:               contents,
+			LocalesToAuthorize: []string{p.SourceLocale},
+			FileType:           sdkfile.FileTypePlaintext,
+			Smartling: struct {
+				Namespace   string
+				FileCharset string
+				Directives  map[string]string
+			}{
+				Directives: p.Directives,
+			},
+		}
+		uploadFileResponse, err := s.uploader.UploadFile(p.AccountUID, p.ProjectID, request)
```

Comment:
https://api-reference.smartling.com/#tag/File-Machine-Translations-(MT)/operation/fileUpload
there is no `ProjectID`